src/main/java/seedu/address/model/diet/DietCollection.java

```diff
@@ -0,0 +1,120 @@
+package seedu.address.model.diet;
+
+//@@author yuntongzhang
+
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Represents a Collection of dietary requirement for a patient.
+ * Guarantees: field values are validated and immutable.
+ */
+public class DietCollection {
+    private final Set<Diet> dietSet;
+
+    public DietCollection() {
+        this.dietSet = new HashSet<>();
+    }
+
+    public DietCollection(Set<Diet> dietSet) {
+        Objects.requireNonNull(dietSet);
+        this.dietSet = dietSet;
```

Comment:
Consider defensive programming here (e.g. create a defensive copy of the incoming dietSet)

Reply:
I will make another PR for this.